flux-refineck/src/checker.rs

```diff
@@ -923,6 +927,37 @@ impl<'a, 'tcx, P: Phase> Checker<'a, 'tcx, P> {
     }
 }
 
+fn type_of_shift<'a>(
+    num_ty: &'a Ty,
+    shift_by_ty: &'a Ty,
+    op: mir::BinOp,
+) -> (BaseTy, &'a rty::RefineArg, &'a rty::RefineArg, sigs::Sig<2>) {
+    match (num_ty.kind(), shift_by_ty.kind()) {
+        (Int!(num_base_ty, num_idxs), Int!(_, shift_by_idxs) | Uint!(_, shift_by_idxs)) => {
+            (
+                BaseTy::Int(*num_base_ty),
+                num_idxs.nth(0),
+                shift_by_idxs.nth(0),
+                sigs::signed_bin_ops(op),
+            )
+        }
+        (Uint!(num_base_ty, num_idxs), Int!(_, shift_by_idxs) | Uint!(_, shift_by_idxs)) => {
```

Comment:
Same here we call this a `uint_ty`

```suggestion
        (Uint!(num_uint_ty, num_idxs), Int!(_, shift_by_idxs) | Uint!(_, shift_by_idxs)) => {
```